# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/svg/SVGIntegrationUtils.cpp
# Commit: dd7fd8b1e328
# Full Hash: dd7fd8b1e328d5f168fa33f692c1e9a79ba633b1
# Author: Jeff Muizelaar <jmuizelaar@mozilla.com>
# Date: 2022-04-23 21:27:33
# Description:
#   Bug 1764943. Add null check on path. r=mstange
#   
#   This problem was introduced by bug 1756197. The old code checked
#   for null before setting the path that was implicitly used by the fill.
#   This null check is retained at the other usage of
# ==============================================================================

diff -r 09a4ccbe6d74 -r dd7fd8b1e328 layout/svg/SVGIntegrationUtils.cpp
--- a/layout/svg/SVGIntegrationUtils.cpp	Fri Apr 22 19:31:11 2022 +0000
+++ b/layout/svg/SVGIntegrationUtils.cpp	Fri Apr 22 20:16:30 2022 +0000
@@ -761,8 +761,10 @@
       ctx.SetDeviceColor(DeviceColor::MaskOpaqueWhite());
       RefPtr<Path> path = CSSClipPathInstance::CreateClipPathForFrame(
           ctx.GetDrawTarget(), frame, mat);
-      ctx.SetPath(path);
-      ctx.Fill();
+      if (path) {
+        ctx.SetPath(path);
+        ctx.Fill();
+      }
 
       return true;
     }
